### PR TITLE
Add support for single operator nodes in YAML dataflow specification

### PR DESCRIPTION
**Pull request by phil-opp**
Single operator nodes are specified through an `operator` field, in contrast to the `operators` list of normal runtime nodes. Since there is only a single operator on such nodes, specifying an additional operator ID is not necessary. If no operator ID is given, the `coordinator` will assign the ID `op`.

Other nodes can also reference outputs of single-operator nodes by only specifying the node ID, the operator ID is optional here as well. The coordinator will automatically resolve these links.

This feature is useful for Python operators because all operators on the same node use a single global interpreter lock (GIL). By using a separate node for each Python operator, they run in different processes without GIL contention.

### CODE REVIEW

File: binaries/runtime/src/operator/mod.rs [hunk 1]
```diff
@@ -9,17 +9,17 @@ mod shared_lib;
 
 pub struct Operator {
     operator_task: Sender<OperatorInput>,
-    config: OperatorConfig,
+    config: OperatorDefinition,
```
Comment (haixuanTao) [inline binaries/runtime/src/operator/mod.rs:12]:
```suggestion
    definition: OperatorDefinition,
```
I feel like doing a search and replace of `config` to `definition` might be nice as it is slightly confusing to read `config().config`

File: binaries/runtime/src/operator/mod.rs [hunk 2]
```diff
@@ -9,17 +9,17 @@ mod shared_lib;
...
 }
 
 impl Operator {
     pub async fn init(
-        operator_config: OperatorConfig,
+        operator_config: OperatorDefinition,
```
Comment (haixuanTao) [inline binaries/runtime/src/operator/mod.rs:17]:
```suggestion
        operator_definition: OperatorDefinition,
```

In the same way, `s/operator_config/operator_definition`

File: binaries/runtime/src/operator/mod.rs [hunk 3]
```diff
@@ -54,7 +54,7 @@ impl Operator {
 
     /// Get a reference to the operator's config.
     #[must_use]
-    pub fn config(&self) -> &OperatorConfig {
+    pub fn config(&self) -> &OperatorDefinition {
```
Comment (haixuanTao) [inline binaries/runtime/src/operator/mod.rs:57]:
```suggestion
    pub fn definition(&self) -> &OperatorDefinition {
```

File: binaries/runtime/src/operator/mod.rs (hunk 1 above)
Comment (phil-opp) [reply binaries/runtime/src/operator/mod.rs]: Done in 2750ea59b1114bc424875b8e80738b154d02b1ff

**Review by haixuanTao — APPROVED** [review]

**Pull request merged**